src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/mapping/Document.java

```diff
@@ -21,4 +21,6 @@
     String ru() default Constants.DEFAULT_REQUEST_UNIT;
 
     int timeToLive() default Constants.DEFAULT_TIME_TO_LIVE;
+
+    boolean autoCreateCollection() default Constants.DEFAULT_AUTO_CREATE_COLLECTION;
```

Comment:
Collection API on document seems odd. Can it be moved to Collection instead?

Reply:
It already has all the collection related annotations, example collection name, ru and time to live.

Reply:
Yeah I agree this is confusing. But this is an annotation which is used for an Entity definition in spring (Document in cosmos). There is no explicit place for Collection in spring-data-cosomsdb world.